lib/Conversion/CombToCGGI/CombToCGGI.cpp

```diff
@@ -115,24 +164,20 @@ struct ConvertTruthTableOp : public OpConversionPattern<TruthTableOp> {
     }
 
     MLIRContext *ctx = getContext();
-    bool ciphertextArg =
-        std::any_of(adaptor.getOperands().begin(), adaptor.getOperands().end(),
-                    [&](const Value &val) {
-                      return isa<lwe::LWECiphertextType>(val.getType()) ||
-                             isa<secret::SecretType>(val.getType());
-                    });
+    bool ciphertextArg = std::any_of(
+        adaptor.getOperands().begin(), adaptor.getOperands().end(),
+        [&](const Value &val) { return isCiphertextOrSecret(val.getType()); });
 
     SmallVector<mlir::Value, 4> lutInputs;
     for (Value val : adaptor.getOperands()) {
       auto integerTy = dyn_cast<IntegerType>(val.getType());
       // If any of the arguments to the truth table are ciphertexts, we must
       // encode and trivially encrypt the plaintext integers arguments.
       if (ciphertextArg && integerTy) {
-        auto encoding = lwe::UnspecifiedBitFieldEncodingAttr::get(
-            ctx, integerTy.getWidth());
+        assert(integerTy.getWidth() == 1 && "LUT inputs should be single-bit");
+        auto ctxtTy = getLWECiphertextForInt(ctx, integerTy, 3);
```

Comment:
Does this 3 also need updating as per the TODO below? If so, maybe add another comment here so we don't miss it later. Or, it can access the type converter (which is stored on the OpConversionPattern) and just reference the bit width field there.

Reply:
Right! Actually I can clean this up. It can't access the type converter's class here without a cast, but I know how to fix that and I'd prefer just calling the type converter directly and hiding this detail too :)

Reply:
Updated.